test/e2e/README.md

```diff
@@ -0,0 +1,26 @@
+# Cross OS E2E Tests
+
+Tests designed to work with both Linux and Windows workers should use the `operating_system` paramater to differentiate differences between operating systems.
+
+Export the `OPERATING_SYSTEM` environment variable to either "linux" or "windows", depending on the OS of the worker that you would like to test.
+
+# OS Specific Tests
+Use the `mark.skipif` paramater to differentiate between operating systems for os specific tests.
+```
+@pytest.mark.skipif(
+        os.environ["OPERATING_SYSTEM"] == "windows",
+        reason="Linux specific test",
+    )
+
+@pytest.mark.skipif(
+        os.environ["OPERATING_SYSTEM"] == "linux",
+        reason="Linux specific test",
+    )
+```
```

Comment:
nit: just to help clarify this  more, what if we change this to:

---

For a Windows-specific test:

```py
@pytest.mark.skipif(
    os.environ["OPERATING_SYSTEM"] == "windows",
    reason="Linux specific test",
)
def test_windows_behaviour() -> None:
    ...
```

For a Linux-specific tests:

```py
@pytest.mark.skipif(
    os.environ["OPERATING_SYSTEM"] == "linux",
    reason="Linux specific test",
)
def test_linux_behaviour() -> None:
    ...
```

Reply:
Updated